packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Dispatch, SetStateAction, useState } from 'react';
+import {
+  useRemoveAnimalBatchesMutation,
+  useRemoveAnimalsMutation,
+} from '../../../store/api/apiSlice';
+import { toLocalISOString } from '../../../util/moment';
+import { parseInventoryId } from '../../../util/animal';
+import { FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+
+const useAnimalOrBatchRemoval = (
+  selectedInventoryIds: string[],
+  setSelectedInventoryIds: Dispatch<SetStateAction<string[]>>,
+) => {
+  const [removeAnimals] = useRemoveAnimalsMutation();
+  const [removeBatches] = useRemoveAnimalBatchesMutation();
+  const [removalModalOpen, setRemovalModalOpen] = useState(false);
+
+  const handleAnimalOrBatchRemoval = (formData: FormFields) => {
+    const timestampedDate = toLocalISOString(formData.date);
+
+    const animalRemovalArray = [];
+    const animalBatchRemovalArray = [];
+
+    for (const id of selectedInventoryIds) {
+      const { kind, id: entity_id } = parseInventoryId(id);
+      if (kind === 'ANIMAL') {
```

Comment:
I think it would be nice if `AnimalOrBatchKeys` is used here.

Reply:
Thank you!!